compiler/semexprs.nim

```diff
@@ -3024,6 +3024,8 @@ proc semExpr(c: PContext, n: PNode, flags: TExprFlags = {}, expectedType: PType
   of nkCall, nkInfix, nkPrefix, nkPostfix, nkCommand, nkCallStrLit:
     # check if it is an expression macro:
     checkMinSonsLen(n, 1, c.config)
+    if n[0].kind == nkIdent and n[0].ident.id == ord(wAuto):
```

Comment:
You don't know at this point that `auto` was not redefined to be a proc overriding `system.auto`. Stop hacking special cases into the compiler. This a fix for `semConv` and it needs to check for than just `auto`. For example `untyped(4)` or `typed("abc)` are likewise illegal constructs.

Reply:
wow,I debug into serveral places haven't found it goes to `semConv`, I'll create a fix in next PR.